src/QuadForm/Quad/ZLattices.jl

```diff
@@ -1392,13 +1393,63 @@ end
 Return whether the row span of `v` lies in the lattice `L`.
 """
 function Base.in(v::fmpq_mat, L::ZLat)
-  @req ncols(v)==degree(L) "The vector should have the same length as the degree of the lattice."
-  @req nrows(v) ==1 "Must be a row vector."
+  @req ncols(v) == degree(L) "The vector should have the same length as the degree of the lattice."
+  @req nrows(v) == 1 "Must be a row vector."
   B = basis_matrix(L)
   fl, w = can_solve_with_solution(B, v, side=:left)
   return fl && isone(denominator(w))
 end
 
+@doc Markdown.doc"""
+    is_primitive(L::ZLat, v::Union{Vector, fmpq_mat})
+                                                   -> Bool
+
+Return whether the vector `v` is primitive in `L`.
+
+A vector `v` in a $\mathbb Z$-lattice `L` is called primitive
+if for all `w` in `L` such that $v = dw$ for some integer `d`,
+then $d = \pm 1$.
+"""
+is_primitive(::ZLat, ::Union{Vector, fmpq_mat})
+
+function is_primitive(L::ZLat, v::Vector{<: RationalUnion})
+  @req v in L "v is not contained in L"
+  M = lattice_in_same_ambient_space(L, matrix(QQ,1,length(v), v))
+  return is_primitive(L, M)
+end
+
+function is_primitive(L::ZLat, v::fmpq_mat)
+  @req v in L "v is not contained in L"
+  M = lattice_in_same_ambient_space(L, v)
+  return is_primitive(L, M)
+end
+
+@doc Markdown.doc"""
+    divisibility(L::ZLat, v::Union{Vector, fmpq_mat}) -> fmpq
+
+Return the divisibility of the primitive vector `v` of `L`.
+
+For a primitive vector `v` in a $\mathbb Z$-lattice `L`, we
+call the divisibility of `v` in `L` the positive generator
+of the $\mathbb Z$-ideal $b(v, L)$, where `b` is the bilinear
+form associated to `L`.
+"""
+divisibility(::ZLat, ::Union{Vector, fmpq_mat})
+
+function divisibility(L::ZLat, v::Vector{<: RationalUnion})
+  @req is_primitive(L, v) "v must be a primitive vector in L"
```

Comment:
No need to assume primitive.
```suggestion
Return the divisibility of `v` with respect to `L`.

For a vector `v` in in the ambient space $(V,\Phi)$ of `L`, we
call the divisibility of `v` with respect to `L` the nonnegative generator
of the fractional $\mathbb Z$-ideal $\Phi(v, L)$.
"""
divisibility(::ZLat, ::Union{Vector, fmpq_mat})

function divisibility(L::ZLat, v::Vector{<: RationalUnion})
```